src/nft/components/profile/view/WalletAssetDisplay.tsx

```diff
@@ -0,0 +1,141 @@
+import { Box } from 'nft/components/Box'
+import { Column } from 'nft/components/Flex'
+import { VerifiedIcon } from 'nft/components/icons'
+import { subheadSmall } from 'nft/css/common.css'
+import { useBag, useIsMobile, useSellAsset } from 'nft/hooks'
+import { WalletAsset } from 'nft/types'
+import { useMemo, useReducer } from 'react'
+import { Link } from 'react-router-dom'
+
+import * as styles from './ProfilePage.css'
+
+const formatEth = (price: number) => {
+  if (price > 1000000) {
+    return `${Math.round(price / 1000000)}M`
+  } else if (price > 1000) {
+    return `${Math.round(price / 1000)}K`
+  } else {
+    return `${Math.round(price * 100 + Number.EPSILON) / 100}`
+  }
+}
+
+export const WalletAssetDisplay = ({ asset, isSellMode }: { asset: WalletAsset; isSellMode: boolean }) => {
+  const sellAssets = useSellAsset((state) => state.sellAssets)
+  const selectSellAsset = useSellAsset((state) => state.selectSellAsset)
+  const removeSellAsset = useSellAsset((state) => state.removeSellAsset)
+  const cartExpanded = useBag((state) => state.bagExpanded)
+  const toggleCart = useBag((state) => state.toggleBag)
+  const isMobile = useIsMobile()
+
+  const [boxHovered, toggleBoxHovered] = useReducer((state) => {
+    return !state
+  }, false)
+  const [buttonHovered, toggleButtonHovered] = useReducer((state) => {
+    return !state
+  }, false)
+
+  const isSelected = useMemo(() => {
+    return sellAssets.some((item) => asset.id === item.id)
+  }, [asset, sellAssets])
+
+  const handleSelect = () => {
+    isSelected ? removeSellAsset(asset) : selectSellAsset(asset)
+    if (
+      !cartExpanded &&
+      !sellAssets.find(
+        (x) => x.tokenId === asset.tokenId && x.asset_contract.address === asset.asset_contract.address
+      ) &&
+      !isMobile
+    )
+      toggleCart()
+  }
+
+  return (
+    <Link
+      to={`/nfts/asset/${asset.asset_contract.address}/${asset.tokenId}?origin=profile`}
```

Comment:
`getAssetHref` in `asset.ts` util can generate this for you

Reply:
I've modified the function to take a WalletAsset as input as well